README.md

```diff
@@ -1,189 +1,180 @@
+x_setter = logged(x_setter, {kind: "setter", isStatic: false});
 ```
 
-The `@defineElement` decorator is based on the `@register` decorator. This decorator is called when the class is finished being defined, and it calls the provided callback with the class that resulted.
-
-```mjs
-// defineElement.mjs
-export decorator @defineElement(name, options) {
-  @register(klass => customElements.define(name, klass, options))
-}
-```
+Note that getters and setters are decorated separately. Accessors are not "coalesced" as in earlier decorators proposals (unless they are generated for a field; see below).
 
-This example uses a `decorator` declaration with a parameter list, which lets a decorator definition take arguments that can be used to supply arguments to other decorators in its definition.
+This desugaring is in terms of the [class static block proposal](https://github.com/tc39/proposal-class-static-block) which exposes a `static { }` construct to be used inside a class body, which runs in the lexical scope of the class. A desugaring in terms of throwaway static private fields would also be possible, but is messy and confusing. However, the decorators proposal does not depend on class static blocks; this is just an explanatory device.
 
-### `@metadata`
+## `@defineElement`
 
-The `@metdata(key, value)` decorator is similar to [`@Reflect.metadata`](https://github.com/rbuckton/reflect-metadata): It allows the easy retrieval of information which was stored by annotating the class. The following examples are written in terms of the Reflect.metadata proposal, but you could imagine storing the metadata in any other form.
+[HTML Custom Elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements) lets you define your own HTML element. Elements are registered using `customElements.define`. Using decorators, the registration can happen up-front:
 
 ```mjs
-import { @metadata } from "./metadata.mjs";
-
-// partially apply the decorator locally for terseness
-decorator @localMeta { @metadata("key", "value") }
+import { defineElement } from "./defineElement.mjs";
 
-@localMeta class C {
-  @localMeta method() { }
-}
-
-Reflect.getMetadata(C, "key");                      // "value"
-Reflect.getMetadata(C.prototype, "key", "method");  // "value"
+@defineElement('my-class')
+class MyClass extends HTMLElement { }
 ```
 
-`@metadata` can also be defined in terms of `@register`. When `@register` is used with a public field, method or accessor, it is called with the second argument being the property key of that class element.
+Classes can be decorated just like methods and accessors. The class shows up in the `value` option.
 
 ```mjs
-// metadata.mjs
-import "reflect-metadata";
-
-export decorator @metadata(key, value) {
-  @register((target, prop) => Reflect.defineMetadata(key, value, target, prop))
+// defineElement.mjs
+export function defineElement(name, options) {
```

Comment:
shouldn't the implementation read:

```javascript
export function defineElement(name) {
   return (klass, options) => {
```

Since `defineElement` is a higher order function that returns a decorator (assuming the `options` here are the generic decorator options, not specific options of `defineElement`)?

Reply:
Oh, I meant options being the custom element definition options. I don't think we need to use the context object for this decorator.

Reply:
Oh, I meant options being the custom element definition options. I don't think we need to use the context object for this decorator.

Reply:
Oh, I meant options being the custom element definition options. I don't think we need to use the context object for this decorator.

Reply:
Oh, I meant options being the custom element definition options. I don't think we need to use the context object for this decorator.